tasks/ssh.yml

```diff
@@ -0,0 +1,179 @@
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*ChallengeResponseAuthentication"
+    line: "ChallengeResponseAuthentication no"
+  notify: Restart sshd
+
+- name: Enable SSH public key authentication in server
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*PubkeyAuthentication"
+    line: "PubkeyAuthentication yes"
+  notify: Restart sshd
+
+- name: Enable SSH public key authentication in client
+  lineinfile:
+    name: /etc/ssh/ssh_config
+    regexp: "^\\s*#?\\s*PubkeyAuthentication"
+    line: "    PubkeyAuthentication yes"
+    insertafter: "^\\s*Host \\*"
+
+- name: Disable weak server keys in SSH client
+  lineinfile:
+    name: /etc/ssh/ssh_config
+    regexp: "^\\s*#?\\s*HostKeyAlgorithms"
+    line: "    HostKeyAlgorithms ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,ssh-rsa"
+    insertafter: "^\\s*Host \\*"
+
+- name: Require group memebership to connect via SSH
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*AllowGroups"
+    line: "AllowGroups ssh-users"
+  notify: Restart sshd
+
+- name: Disable root login via SSH
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*PermitRootLogin"
+    line: "PermitRootLogin no"
+  notify: Restart sshd
+
+- name: Disable weak SSH server ciphers
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*Ciphers"
+    line: "Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr"
+  notify: Restart sshd
+
+- name: Disable weak SSH client ciphers
+  lineinfile:
+    name: /etc/ssh/ssh_config
+    regexp: "^\\s*#?\\s*Ciphers"
+    line: "    Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr"
+    insertafter: "^\\s*Host \\*"
+
+- name: Disable weak SSH server MACs
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*MACs"
+    line: "MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-ripemd160-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,hmac-ripemd160,umac-128@openssh.com"
+  notify: Restart sshd
+
+- name: Enable sshd keepalives
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: "^\\s*#?\\s*ClientAliveInterval"
+    line: "ClientAliveInterval 120"
+  notify: Restart sshd
+
+- name: Disable weak SSH client MACs
+  lineinfile:
+    name: /etc/ssh/ssh_config
+    regexp: "^\\s*#?\\s*MACs"
+    line: "    MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-ripemd160-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,hmac-ripemd160,umac-128@openssh.com"
+    insertafter: "^\\s*Host \\*"
+
+- name: Check if sshd moduli were generated
+  stat:
+    path: /etc/ssh/.moduli-generated
+  register: sshd_moduli
+
+- name: Generate all sshd moduli
+  command: ssh-keygen -G /etc/ssh/moduli.all -b 4096
```

Comment:
Ich weiß nicht, glaube eher, das hier eher in die Zukunft investiert werden sollte, als Kompatibilität für $Szenario zu gewährleisten. Mit fällt da eigentlich auch keines ein, wie sieht der Rest das, übersehe ich da was?